keps/sig-node/1029-ephemeral-storage-quotas/README.md

```diff
@@ -667,30 +651,96 @@ There is no failure history or flakes in https://storage.googleapis.com/k8s-tria
+
+<!--
+Based on reviewers feedback describe what additional tests need to be added prior
+implementing this enhancement to ensure the enhancements have also solid foundations.
+-->
+
+#### Unit tests
+
+The quota code is by an large not very amendable to unit tests.  While
+there are simple unit tests for parsing the mounts file, and there
+could be tests for parsing the projects and projid files, the real
+work (and risk) involves interactions with the kernel and with
+multiple instances of this code (e. g. in the kubelet and the runtime
+manager, particularly under stress).  It also requires setup in the
+form of a prepared filesystem.  It would be better served by
+appropriate end to end tests.
+
+The main unit test is in package under `pkg/volume/util/fsquota/`.
+
+- `pkg/volume/util/fsquota/`: `2024-06-12` - `73.9%`
+- - project.go 75.7%
+- - quota.go 100%
+- - quota_linux.go 72.2%
+
+See details in https://testgrid.k8s.io/sig-testing-canaries#ci-kubernetes-coverage-unit&include-filter-by-regex=fsquota.
+
+#### Integration tests
+
+N/A
+
+#### e2e tests
+
+e2e evolution (LocalStorageCapacityIsolationQuotaMonitoring [Slow] [Serial] [Disruptive] [Feature:LocalStorageCapacityIsolationQuota][NodeFeature:LSCIQuotaMonitoring]) can be found in [`test/e2e_node/quota_lsci_test.go`](https://github.com/kubernetes/kubernetes/blob/8cd689e40d253e520b1698d4bcf33992f0ae1d20/test/e2e_node/quota_lsci_test.go#L93-L103)
+
+The e2e tests are slow and serial and we will not promote them to be conformance test then.
+There is no failure history or flakes in https://storage.googleapis.com/k8s-triage/index.html?test=LocalStorageCapacityIsolationQuotaMonitoring
+
+
+### Graduation Criteria
 
 The following criteria applies to
 `LocalStorageCapacityIsolationFSMonitoring`:
 
-### Phase 1: Alpha (1.15)
+#### Alpha
 
 - Support integrated in kubelet
 - Alpha-level documentation
 - Unit test coverage
 - Node e2e test
 
-### Phase 2: Beta (v1.29)
+#### Beta
 
 - User feedback
 - Benchmarks to determine latency and overhead of using quotas
   relative to existing monitoring solution
 - Cleanup
+- Use Ephemeral-Storage-Quotas in User Namespace
 
-### Phase 3: GA
+#### GA
 
 - TBD
 
-## Performance Benchmarks
+### Upgrade / Downgrade Strategy
+
+Turn off the feature gate to turn off the feature.
+
+### Version Skew Strategy
+
+#### Kubelet and API Server Skew:
+
+1. If the API server is on the latest version with the user namespace feature flag enabled, 
+and the kubelet also has the user namespace feature along with the 
+LocalStorageCapacityIsolationFSQuotaMonitoring feature flag enabled, Pods with hostUsers 
+set to false will have XFS quotas within user namespaces enabled.
+2. If any of the necessary feature flags (user namespace on either the API server or kubelet,
+or LocalStorageCapacityIsolationFSQuotaMonitoring on the kubelet) are not enabled, then, 
+XFS quotas within user namespaces will not be supported
+
+
+### Performance Benchmarks
```

Comment:
Nit: this entire section seems best suited in the testing section, I'd just move all of that after `##### e2e tests`.